src/main/java/de/tum/in/www1/artemis/service/connectors/bamboo/BambooService.java

```diff
@@ -566,11 +570,17 @@ public Result onBuildCompleted(ProgrammingExerciseParticipation participation, O
             // Do not remove this save, otherwise Hibernate will throw an order column index null exception on saving the build logs
             programmingSubmission = programmingSubmissionRepository.save(programmingSubmission);
 
+            // todo wip, save currently necessary, because of relationship between result and submission
+            result = resultRepository.save(result);
+
             var buildLogs = extractAndPrepareBuildLogs(buildResult, programmingSubmission);
             // Set the received logs in order to avoid duplicate entries (this removes existing logs)
+            result.setSubmission(programmingSubmission);
+            programmingSubmission.setResult(result);
+            programmingSubmission = programmingSubmissionRepository.save(programmingSubmission);
```

Comment:
we now save the programmingSubmission twice (see two lines below), is this really necessary?

Reply:
Without the second save there would be an IndexColumnError null thrown for the build logs. We tried creating a BuildLogEntryRepository and first saving all BuildLogEntries, before establishing the relation but that did not fix the problem. This was the only solution @SimonEntholzer found. Maybe you can explain it a bit better?

Reply:
You were right, this was not needed. There was still a issue with `null index column` for `buildLogEntries` which is fixed in `6444cd`.